src/parser/mod.rs

```diff
@@ -103,6 +109,67 @@ pub struct KeyMetadata {
     pub size: u32,
 }
 
+/// A trait for parsing multiple types of values to memcached values.
+pub trait ToMemcachedValue {
```

Comment:
Is the intention that we want library consumers to be able to implement this on their own types? My preference would be to not allow that unless there is a need, and to prevent that with a private supertrait as described [here](https://predr.ag/blog/definitive-guide-to-sealed-traits-in-rust/).

Something like:
```rust
mod private {
  pub(super) trait ToMemcachedValue {
    ...
  }
}

pub trait ToMemcachedValue: private::ToMemcachedValue {
  ...
}

Reply:
Thanks, I've refactored this with the sealed trait pattern.